dvc/remote/base.py

```diff
@@ -942,23 +967,33 @@ def _checkout(
         force=False,
         progress_callback=None,
         relink=False,
+        filter_info=None,
     ):
         if not self.is_dir_checksum(checksum):
             return self._checkout_file(
                 path_info, checksum, force, progress_callback=progress_callback
             )
         return self._checkout_dir(
-            path_info, checksum, force, progress_callback, relink
+            path_info, checksum, force, progress_callback, relink, filter_info
         )
 
-    def get_files_number(self, checksum):
+    def get_files_number(self, path_info, checksum, filter_info):
+        from funcy.py3 import lfilter
+
         if not checksum:
             return 0
 
-        if self.is_dir_checksum(checksum):
+        if not self.is_dir_checksum(checksum):
+            return 1
+
+        if not filter_info:
             return len(self.get_dir_cache(checksum))
 
-        return 1
+        def _func(entry):
+            info = path_info / entry[self.PARAM_RELPATH]
+            return filter_info == info or info.isin(filter_info)
```

Comment:
Looks like we use this `== or isin` a lot all over the code. Maybe something like `path_info.matches`? It would also be useful for globs and stuff later.

Reply:
`isineq` or `isin_or_eq` )